src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -1890,6 +1888,8 @@ else if ((v = w) == null)
                 }
                 break;
             }
+            if (q != null && q.base - qbase > 0)
+                break;
```

Comment:
Crazy idea: Only create new workers on successful CAS if no queue-drift has occurred, but always prioritize to wake up sleeping workers regardless?

```suggestion
                if (v != null) {
                    v.phase = sp;
                    if (v.parking != 0)
                        U.unpark(v.owner);
                } else if (q == null || q.base - qbase == 0)
                    createWorker();

                break;
            }
            if (q != null && q.base - qbase > 0)
                break;
```

Reply:
Narrator: Turns out it was crazy, as it requires a back-out of nc. Disregard the comment above :)